libs/data-mapper/src/lib/utils/Connection.Utils.ts

```diff
@@ -40,152 +47,138 @@ export const createConnectionEntryIfNeeded = (
   }
 };
 
-// Primarily used for drawn connections
-export const addNodeToConnections = (
+/**
+ * Creates or updates a connection entry's input value.
+ * This is the gateway for *setting ANY input values* in dataMapConnections - including
+ * both those that are deserialized, handle-drawn, or InputDropdown-set.
+ *
+ * @param connections The connections dictionary to add the entry to
+ * @param targetNode The node to create the connection entry for
+ * @param targetNodeReactFlowKey The reactFlowKey of `targetNode`
+ * @param inputIndex The index of the input to set the value for (only needed if NOT `isHandleDrawnOrDeserialized`)
+ * @param value The value to set the input to (InputConnection, or `null` to delete an unbounded input)
+ * @param isHandleDrawnOrDeserialized Whether the new connection being made is handle-drawn or a deserialization
```

Comment:
Instead of the motion causing it, maybe this could be named after the action that results from it? Or if that doesn't make sense, what about the inverse so it's not named "is A Or B", but is named "is C" and we worry about the neg case